app/src/main/java/com/x8bit/bitwarden/data/platform/repository/DebugMenuRepositoryImpl.kt

```diff
@@ -0,0 +1,70 @@
+package com.x8bit.bitwarden.data.platform.repository
+
+import com.x8bit.bitwarden.BuildConfig
+import com.x8bit.bitwarden.data.platform.datasource.disk.FeatureFlagOverrideDiskSource
+import com.x8bit.bitwarden.data.platform.manager.model.FlagKey
+import com.x8bit.bitwarden.data.platform.repository.util.bufferedMutableSharedFlow
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.onSubscription
+import kotlinx.serialization.json.JsonPrimitive
+
+/**
+ * Default implementation of the [DebugMenuRepository]
+ */
+class DebugMenuRepositoryImpl(
+    private val featureFlagOverrideDiskSource: FeatureFlagOverrideDiskSource,
+    private val serverConfigRepository: ServerConfigRepository,
+) : DebugMenuRepository {
+
+    private val mutableOverridesUpdatedFlow = bufferedMutableSharedFlow<Unit>(replay = 1)
+    override val featureFlagOverridesUpdatedFlow: Flow<Unit> = mutableOverridesUpdatedFlow
+        .onSubscription { emit(Unit) }
+
+    override val isDebugMenuEnabled: Boolean
+        get() = BuildConfig.HAS_DEBUG_MENU
+
+    override fun <T : Any> updateFeatureFlag(key: FlagKey<T>, value: T) {
+        featureFlagOverrideDiskSource.saveFeatureFlag(key = key, value = value)
+        mutableOverridesUpdatedFlow.tryEmit(Unit)
+    }
+
+    override fun <T : Any> getFeatureFlag(key: FlagKey<T>): T? =
+        featureFlagOverrideDiskSource.getFeatureFlag(
+            key = key,
+        )
+
+    override suspend fun resetFeatureFlagOverrides() {
+        serverConfigRepository
+            .getServerConfig(
+                forceRefresh = true,
```

Comment:
Do we need this to actually refresh?

I am mainly thinking that we could just use the `serverConfigStateFlow.value` flag an not have to deal with the suspending stuff.

Reply:
server config gives me (after drilling down) the <String, JsonPrimitive> map... no thanks! Would this sound good to you 
```
return debugMenuRepository.getFeatureFlag(key)
            ?: defaultFeatureFlagManager.getFeatureFlag(
                key,
                forceRefresh,
            )
```
In this case we would only call the suspending function while an override value has not been set.

Reply:
PEBKAC